HomeWork18/HomeWork18/App.cs

```diff
@@ -5,45 +5,60 @@ namespace HomeWork18
     internal class App
     {
         private readonly IUserService _userService;
+        private readonly IResourseService _resourseService;
+        private readonly IAuthenticationService _authenticationService;
 
-        public App(IUserService userService)
+        public App(
+            IUserService userService,
+            IResourseService resourseService,
+            IAuthenticationService authenticationService)
         {
             _userService = userService;
+            _resourseService = resourseService;
+            _authenticationService = authenticationService;
         }
 
         public async Task StartUp()
         {
-            var users = await _userService.GetListUsersAsync(2);
+            var getUsers = Task.Run(async () => await _userService.GetUsersAsync(2));
+            var getUser = Task.Run(async () => await _userService.GetUserAsync(7));
+            var getFailedUser = Task.Run(async() => await _userService.GetUserAsync(23));
 
-            var user = await _userService.GetUserAsync(7);
+            var getSource = Task.Run(async () => await _resourseService.GetResourceAsync(2));
+            var getSources = Task.Run(async () => await _resourseService.GetListResourcesAsync());
+            var getFailedSource = Task.Run(async () => await _resourseService.GetResourceAsync(23));
 
-            user = await _userService.GetUserAsync(23);
+            var CreateEmployee = Task.Run(async () => await _userService.CreateEmployeeAsync("morpheus", "leader"));
+            var updateEmployeePut = Task.Run(async () => await _userService.UpdateEmployeeAsync(2, "morpheus", "zion resident"));
+            var modifyEmployeePath = Task.Run(async () => await _userService.ModifyEmployeeAsync(2, "morpheus", "zion resident"));
+            var deleteEmployee = Task.Run(async () => await _userService.DeleteEmployee(2));
 
-            var source = await _userService.GetResourceAsync(1);
+            var register = Task.Run(async () => await _authenticationService.RegisterAsync("eve.holt@reqres.in", "pistol"));
+            var registerFailed = Task.Run(async () => await _authenticationService.RegisterAsync("sydney@fife", null!));
+            var login = Task.Run(async () => await _authenticationService.LoginAsync("eve.holt@reqres.in", "cityslicka"));
+            var loginFailed = Task.Run(async () => await _authenticationService.LoginAsync("peter@klaven", null!));
```

Comment:
For what you need all these?

Reply:
for asynhron work all this implementation requests